app/controllers/applications_controller.rb

```diff
@@ -1,6 +1,9 @@
 class ApplicationsController < ApplicationController
   def show
-    @applications = Application.find(params[:id])
+    @application = Application.find(params[:id])
+    return unless params[:pet_name]
```

Comment:
Interesting conditional use.

Reply:
It's good to know how much functionality we can put into these controllers, I think i thought that we could only put in what it has access to, not realizing their could be conditional logic in the controller as well.

Reply:
Exactly! It is nice to know that our logic knowledge as related to Ruby can help in constructing a logical data flow.

Reply:
@pets = Pet.search(params[:pet_name]) if params[:pet_name]

Returning early can make the code wonky later on.